src/fenics_constitutive/stress_strain.py

```diff
@@ -132,9 +138,198 @@ def strain_from_grad_u(grad_u: np.ndarray, constraint: Constraint) -> np.ndarray
-            strain_view[:, 4] = 1 / 2**0.5 * (grad_u_view[:, 5] + grad_u_view[:, 7])
-            strain_view[:, 5] = 1 / 2**0.5 * (grad_u_view[:, 2] + grad_u_view[:, 6])
+            strain_view[:, 4] = 1 / 2**0.5 * (grad_u_view[:, 2] + grad_u_view[:, 6])
+            strain_view[:, 5] = 1 / 2**0.5 * (grad_u_view[:, 5] + grad_u_view[:, 7])
         case _:
             msg = "Constraint not supported."
             raise NotImplementedError(msg)
     return strain
+
+
+class UniaxialStrainFrom3D(IncrSmallStrainModel):
+    """
+    Convert a 3D model to a uniaxial strain model. This is achieved by copying the
+    relevant 1D components to a 3D array, calling the 3D model, and then copying the
+    3D components back to the 1D array. Since this converter creates new arrays for
+    the 3D components, it is (probably) not suitable for large-scale simulations due
+    to the memory consumption.
+
+    Args:
+        model: 3D model to convert to uniaxial strain.
+
+    Attributes:
+        model: 3D model to convert to uniaxial strain.
+        stress_3d: 3D stress array.
+        tangent_3d: 3D tangent array.
+        grad_del_u_3d: 3D array of gradient of displacement increment.
+    """
+
+    def __init__(self, model: IncrSmallStrainModel) -> None:
+        assert model.constraint == Constraint.FULL
+        self.model = model
+        self.stress_3d = None
+        self.tangent_3d = None
+        self.grad_del_u_3d = None
+
+    @property
+    def constraint(self) -> Constraint:
+        return Constraint.UNIAXIAL_STRAIN
+
+    def update(self) -> None:
+        self.model.update()
+
+    def evaluate(
+        self,
+        time: float,
+        del_t: float,
+        grad_del_u: np.ndarray,
+        mandel_stress: np.ndarray,
+        tangent: np.ndarray,
+        history: np.ndarray | dict[str, np.ndarray] | None,
+    ) -> None:
+        self.tangent_3d = (
+            np.zeros(6 * 6 * len(grad_del_u))
+            if self.tangent_3d is None
+            else self.tangent_3d
+        )
+        self.stress_3d = (
+            np.zeros(6 * len(grad_del_u)) if self.stress_3d is None else self.stress_3d
+        )
+        self.grad_del_u_3d = (
+            np.zeros(9 * len(grad_del_u))
+            if self.grad_del_u_3d is None
+            else self.grad_del_u_3d
+        )
+
+        self._grad_del_u_to_3d(grad_del_u, self.grad_del_u_3d)
+        self._mandel_vector_to_3d(mandel_stress, self.stress_3d)
+
+        self.model.evaluate(
+            time, del_t, self.grad_del_u_3d, self.stress_3d, self.tangent_3d, history
+        )
+        self._tangent_to_1d(self.tangent_3d, tangent)
+        self._mandel_vector_to_1d(self.stress_3d, mandel_stress)
+
+    @property
+    def history_dim(self) -> dict[str, int | tuple[int, int]] | None:
+        return self.model.history_dim
+
+    @df.common.timed("model-conversion-wrapper")
+    def _grad_del_u_to_3d(
```

Comment:
Don't understand the fct.
It's a class function and the parameter you overwrite (grad_del_u_3d) is a class variable so why not:

    ```def _grad_del_u_to_3d(
        self, grad_del_u_1d: np.ndarray ) -> None:
        self.grad_del_u_3d.reshape(-1, 9)[:, 0] = grad_del_u_1d```

??
Same for the following fcts